### DISCUSSION TITLE
Payment connector extension not connecting

### DISCUSSION THREAD

**Discussion by Ashley-Bode** [microsoft/Dynamics365Commerce.Solutions#297] [Q&A]
We are in the process of implementing our payment connector in CommerceSDK from RetailSDK however the connection cannot be established. We have followed guides from
https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/end-to-end-payment-extension
https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/end-to-end-payment-extension#configure-the-payment-connector-in-the-hardware-station-config
 
Nothing so far has resulted in establishing the connection. We also noted that INamedRequestHandler is Obsolete and INamedRequestHandlerAsync is required however the change has not affected the end result. Our RetailSDK implementation is working, all we require is some assistance in getting the CommerceSDK implementation to work.

The error we receive on our POS is:
Error occurred while communicating with Hardware Station : {"ErrorResourceId":"Microsoft_Dynamics_Commerce_HardwareStation_PaymentTerminal_Error","LocalizedMessage":"There was an error communicating with the payment device. Check the device or see your system administrator.","ClassName":"Microsoft.Dynamics.Commerce.HardwareStation.PeripheralException","Message":"Payment terminal 'Connector', specified in the Store's Hardware Profile, could not be loaded.\n\nThis can take place if the Payment Terminal Device assembly is not:\n- available in the directory specified in the diagnostics event corresponding to the ID 7124;\n- listed in the section hardwareStation->composition of the configuration file HardwareStation.Dedicated/Shared.config located in the above directory;\n- exporting a type annotated with Connector

**Comment by mica1900** [comment]
Hi,

Can you find out, via the event viewer, which directory is ment in the error message "......Payment Terminal Device assembly is not:\n- available in the directory specified........."?

I assume you also have a payment processor (IPaymentprocessor) right? We copy that to the following locations:
K:\AOSService\PackagesLocalDirectory\Bin\Connectors
K:\AOSService\WebRoot\Bin
K:\RetailServer\webroot\bin
K:\RetailServer\webroot\bin\Connectors
K:\RetailServer\webroot\bin\Ext
C:\Program Files\Microsoft Dynamics 365\10.0\Store Commerce\Extensions\RanzijnCustomizations\HardwareStation
K:\Workfolder\Customizations\HardwareStation (My hardwarestation folder from my VS project)

Further the payment connector (INamedRequestHandler) needs to exist in the Visual Studio hardware station project and in the 'PIN Pad' section of the hardwareprofile you need to setup your driver and device name.

Hope this helps.

**Comment by etch104** [comment]
Hello I was facing the same issue locating the below path 
C:\Program Files\Microsoft Dynamics 365\10.0\Commerce hardware station\Microsoft\Webroot

Copy the extension DLL inside the bin folder and the above path and the ET folder

![image](https://github.com/user-attachments/assets/82c6cd37-191b-4fab-95bb-152403ba027a)


Make sure to set the configuration DlLS
